trace/core/src/main/scala/org/http4s/otel4s/middleware/trace/HeadersAllowedAsAttributes.scala

```diff
@@ -0,0 +1,191 @@
+    union(that)
+}
+
+object HeadersAllowedAsAttributes {
+
+  /** @return an instance where the same set of headers is allowed for both
+    *         requests and responses
+    */
+  def apply(allowed: Set[CIString]): HeadersAllowedAsAttributes =
+    apply(allowed, allowed)
+
+  /** The default sets of HTTP headers allowed to be turned into `Attribute`s. */
+  lazy val default: HeadersAllowedAsAttributes =
+    apply(defaultAllowedHttpHeaders)
+
+  /** No headers are allowed to be made into `Attribute`s. */
+  val none: HeadersAllowedAsAttributes = apply(Set.empty)
+
+  private[this] lazy val defaultAllowedHttpHeaders: Set[CIString] = Set(
+    "Accept",
+    "Accept-CH",
+    "Accept-Charset",
+    "Accept-CH-Lifetime",
+    "Accept-Encoding",
+    "Accept-Language",
+    "Accept-Ranges",
+    "Access-Control-Allow-Credentials",
+    "Access-Control-Allow-Headers",
+    "Access-Control-Allow-Origin",
+    "Access-Control-Expose-Methods",
+    "Access-Control-Max-Age",
+    "Access-Control-Request-Headers",
+    "Access-Control-Request-Method",
+    "Age",
+    "Allow",
+    "Alt-Svc",
+    "B3",
+    "Cache-Control",
+    "Clear-Site-Data",
+    "Connection",
+    "Content-Disposition",
+    "Content-Encoding",
+    "Content-Language",
+    "Content-Length",
+    "Content-Location",
+    "Content-Range",
+    "Content-Security-Policy",
+    "Content-Security-Policy-Report-Only",
+    "Content-Type",
+    "Cross-Origin-Embedder-Policy",
+    "Cross-Origin-Opener-Policy",
+    "Cross-Origin-Resource-Policy",
+    "Date",
+    "Deprecation",
+    "Device-Memory",
+    "DNT",
+    "Early-Data",
+    "ETag",
+    "Expect",
+    "Expect-CT",
+    "Expires",
+    "Feature-Policy",
+    "Forwarded",
+    "From",
+    "Host",
+    "If-Match",
+    "If-Modified-Since",
+    "If-None-Match",
+    "If-Range",
+    "If-Unmodified-Since",
+    "Keep-Alive",
+    "Large-Allocation",
+    "Last-Modified",
+    "Link",
+    "Location",
+    "Max-Forwards",
+    "Origin",
+    "Pragma",
+    "Proxy-Authenticate",
+    "Public-Key-Pins",
+    "Public-Key-Pins-Report-Only",
+    "Range",
+    "Referer",
+    "Referer-Policy",
+    "Retry-After",
+    "Save-Data",
+    "Sec-CH-UA",
+    "Sec-CH-UA-Arch",
+    "Sec-CH-UA-Bitness",
+    "Sec-CH-UA-Full-Version",
+    "Sec-CH-UA-Full-Version-List",
+    "Sec-CH-UA-Mobile",
+    "Sec-CH-UA-Model",
+    "Sec-CH-UA-Platform",
+    "Sec-CH-UA-Platform-Version",
+    "Sec-Fetch-Dest",
+    "Sec-Fetch-Mode",
+    "Sec-Fetch-Site",
+    "Sec-Fetch-User",
+    "Server",
+    "Server-Timing",
+    "SourceMap",
+    "Strict-Transport-Security",
+    "TE",
+    "Timing-Allow-Origin",
+    "Tk",
+    "Trailer",
+    "Transfer-Encoding",
+    "Upgrade",
+    "User-Agent",
+    "Vary",
+    "Via",
+    "Viewport-Width",
+    "Warning",
+    "Width",
+    "WWW-Authenticate",
+    "X-B3-Sampled",
+    "X-B3-SpanId",
+    "X-B3-TraceId",
+    "X-Content-Type-Options",
```

Comment:
Should we add the X-Correlation-Id on here as we already have the X-Request-Id?

Reply:
yup, adding it. the list was just copied from the natchez middleware originally, in case that wasn't clear